kv.go

```diff
@@ -138,6 +142,51 @@ func (k *KVService) CompareAndDelete(key string, oldValue interface{}) (bool, er
 	return k.Set(key, nil, SetAtomic(oldValue))
 }
 
+// SetAtomicWithRetries will set a key-value pair atomically using compare and set semantics:
+// it will read key's value (to get oldValue), perform valueFunc (to get newValue),
+// and compare and set (comparing oldValue and setting newValue).
+//
+// Parameters:
+//
+//      `key`              is the key to get and set.
+//      `valueFunc`        is a user-provided function that will take the old value as a []byte and
+//                         return the new value or an error. If valueFunc needs to operate on
+//                         oldValue, it will need to use the oldValue as a []byte, or convert
+//                         oldValue into the expected type (e.g., by parsing it, or marshaling it
+//                         into the expected struct). It should then return the newValue as the type
+//                         expected to be stored.
+//
+// Returns:
+//
+//       Returns err if the key could not be retrieved (DB error), valueFunc returned an error,
+//               if the key could not be set (DB error), or if the key could not be set (after retries).
+//       Returns nil if the value was set.
+//
+// Minimum server version: 5.18
+func (k *KVService) SetAtomicWithRetries(key string, valueFunc func(oldValue []byte) (newValue interface{}, err error)) error {
```

Comment:
any reason why `oldValue` is `[]byte` but `newValue` is `interface{}`? should they both be `interface{}`?

Reply:
oh just saw your other comment :D